packages/fx-core/tests/core/middleware/projectMigrationV3.test.ts

```diff
@@ -1082,6 +1084,24 @@ describe("userdataMigration", () => {
     );
     assert.equal(testEnvContent_local, trueEnvContent_local);
   });
+
+  it("Should successfully resolve different EOLs of userdata", async () => {
+    sandbox
+      .stub(fs, "readFileSync")
+      .returns(
+        "fx-resource-aad-app-for-teams.clientSecret=abcd\nfx-resource-bot.botPassword=1234\n"
+      );
+
+    const migrationContext = await mockMigrationContext(projectPath);
+    await copyTestProject(Constants.happyPathTestProject, projectPath);
+    await userdataMigration(migrationContext);
+
```

Comment:
can we call sandbox.restore() here? The current code works fine, my only concern is the mock may impact `assertFileContent()` function if someone changed its implementation to use `readFileSync()`. Restore the mocked function in advance can avoid such kind of issues.

Reply:
reasonable concern. modified in the latest commit